source/extensions/filters/network/mysql_proxy/mysql_utils.cc

```diff
@@ -216,6 +226,129 @@ DecodeStatus BufferHelper::peekHdr(Buffer::Instance& buffer, uint32_t& len, uint
   return DecodeStatus::Success;
 }
 
+AuthMethod AuthHelper::authMethod(uint32_t cap, const std::string& auth_plugin_name) {
+  if (auth_plugin_name.empty()) {
+    /*
+     * https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase.html
+     */
+    bool v41 = cap & CLIENT_PROTOCOL_41;
+    bool sconn = cap & CLIENT_SECURE_CONNECTION;
+    if (!v41 || !sconn) {
+      return AuthMethod::OldPassword;
+    }
+    return AuthMethod::NativePassword;
+  }
+
+  if (auth_plugin_name == MYSQL_OLD_PASSWORD) {
+    return AuthMethod::OldPassword;
+  }
+  if (auth_plugin_name == MYSQL_NATIVE_PASSWORD) {
+    return AuthMethod::NativePassword;
+  }
+  if (auth_plugin_name == MYSQL_SHA256_PASSWORD) {
+    return AuthMethod::Sha256Password;
+  }
+  if (auth_plugin_name == MYSQL_CACHE_SHA_PASSWORD) {
+    return AuthMethod::CacheSha2Password;
+  }
+  if (auth_plugin_name == MYSQL_CLEAR_PASSWORD) {
+    return AuthMethod::ClearPassword;
+  }
+  return AuthMethod::Unknown;
+}
+
+// https://github.com/mysql/mysql-server/blob/5.5/sql/password.c#L186
+std::vector<uint8_t> OldPassword::signature(const std::string& password,
+                                            const std::vector<uint8_t>& seed) {
+  std::vector<uint8_t> to;
+  if (password.empty()) {
+    return to;
+  }
+  to.resize(SEED_LENGTH);
+  auto hash_pass = hash(password);
+  auto hash_seed = hash(seed);
+  RandStruct rand_st(hash_pass[0] ^ hash_seed[0], hash_pass[1] ^ hash_seed[1]);
+  for (int i = 0; i < SEED_LENGTH; i++) {
+    to[i] = static_cast<uint8_t>((floor(rand_st.myRnd() * 31) + 64));
+  }
+  char extra = static_cast<char>(floor(rand_st.myRnd() * 31));
```

Comment:
Can you make _extra_ to be _uint8_t_ for consistency?

Reply:
fixed.